File: src/ProjectR/content/scripts/spells/Stonebreaker.lua
function GetTargetType()
    return Single
end

function GetName()
return "Stonebreaker"
end

function GetDescription()
  return "A harsh kick that ignores a bunch of DEF."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return .2
end

function GetDelay()
  return .5
end

function GetSpellType()
  return Physical
end

function GetMasteries()
  return {  }
end

function SpellEffect()
  local damage = 4.5 * aAD - .5 * dDEF
  d:TakeDamage(damage)
end

File: src/ProjectR.Scripting/content/scripts/spells/StoneBreaker.cs
﻿using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting.Spells
{
    public class StoneBreaker : SpellScriptBase
    {
        public override string Name { get { return "Stonebreaker"; } }
        public override string Description { get { return "A harsh kick that ignores a bunch of DEF."; } }

        public override TargetType TargetType { get { return TargetType.Single; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] { }; } }
        public override SpellType SpellType { get { return SpellType.Physical; } }
        public override bool IsSupportSpell { get { return false; } }

        public override double MPCost { get { return .2; } }
        public override double Delay { get { return .5; } }

        public override void SpellEffect(ICharacter caster, ICharacter target)
        {
            var damage = 4.5 * aAD - .5 * dDEF;
            DealDamage(damage);
        }
    }
}